Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs

```diff
@@ -220,6 +156,32 @@ type FSharpKernel() as this =
         |> Seq.filter (fun v -> v <> "it") // don't report special variable `it`
         |> Seq.choose (this.GetCurrentVariable)
 
+    override _.ScriptExtension with get() =
+        DefaultScriptExtension
+
+    override this.AddScriptReferences (packageReferences: IReadOnlyList<ResolvedPackageReference>) =
+        // Generate #r and #I from packageReferences
+        let sb = StringBuilder()
+        let hashset = HashSet()
+
+        for reference in packageReferences do
+            for assembly in reference.AssemblyPaths do
+                if hashset.Add(assembly.FullName) then
+                    if assembly.Exists then
+                        sb.AppendFormat("#r @\"{0}\"", assembly.FullName) |> ignore
+                        sb.Append(Environment.NewLine) |> ignore
+
+            match reference.PackageRoot with
+            | null -> ()
+            | root ->
+                if hashset.Add(root.FullName) then
+                    if root.Exists then
+                        sb.AppendFormat("#I @\"{0}\"", root.FullName) |> ignore
+                        sb.Append(Environment.NewLine) |> ignore
+        let command = new SubmitCode(sb.ToString(), "fsharp")
+        let task = this.SendAsync(command)
+        task.Wait()
```

Comment:
We shouldn't assume there's no `SynchronizationContext`. This could result in a deadlock.

You might consider using `DeferCommand` (which is synchronous) since until the next submission occurs, the script reference has no effect.